Modules/IO/ImageBase/include/itkImageFileWriter.h

```diff
@@ -236,11 +236,10 @@ class ITK_TEMPLATE_EXPORT ImageFileWriter : public ProcessObject
   bool m_UseInputMetaDataDictionary{ true };
 };
 
-
 /** Convenience function for writing an image. */
 template <typename TImage>
 ITK_TEMPLATE_EXPORT void
-WriteImage(const TImage * image, const char * filename, bool compress = false)
+WriteImage(const TImage * image, const std::string & filename, bool compress = false)
 {
   using WriterType = ImageFileWriter<TImage>;
   typename WriterType::Pointer writer = WriterType::New();
```

Comment:
It would be nice to use C++11 auto here: `const auto writer = WriterType::New();`

Reply:
I tried that (based on your original PR), and I remember there were compile errors.

Reply:
Strange! Do you remember what compile errors?